core/domain/classifier_domain.py

```diff
@@ -72,11 +116,52 @@ def to_dict(self):
         """
 
         return {
-            'classifier_id': self.id,
-            'exp_id': self.exp_id,
-            'exp_version_when_created': self.exp_version_when_created,
-            'state_name': self.state_name,
-            'algorithm_id': self.algorithm_id,
-            'cached_classifier_data': self.cached_classifier_data,
-            'data_schema_version': self.data_schema_version
+            'classifier_id': self._id,
+            'exp_id': self._exp_id,
+            'exp_version_when_created': self._exp_version_when_created,
+            'state_name': self._state_name,
+            'algorithm_id': self._algorithm_id,
+            'cached_classifier_data': self._cached_classifier_data,
+            'data_schema_version': self._data_schema_version
         }
+
+    def validate(self):
```

Comment:
If the validate() method is in the domain file then its tests should be in classifier_domain_tests. If this involves moving over some tests from the services file, that's fine, but bear in mind that it's the validate() method that should be being tested and nothing else.